### PR TITLE
Put dependency versions

### LINKED ISSUE #2470: Specify minimum version inter-dependencies for SDK libraries
KeyVault 1.0 (GA) would depend on Core 1.1.0 (GA)
Storage 1.0.1 (GA) would still depend on Core 1.0 (GA)
Identity 1.1.0-beta.1 would still depend on Core 1.0 (GA)

Versions should be specified at leas in the CMakeFiles in `find_package` statements, similar to how we do it for libcurl.

Also, look into the recent vcpkg features, if they have a way to also specify that info in manifest files (if `version>=` is available / does not fail to install when customers use vcpkg without any additional flags - https://devblogs.microsoft.com/cppblog/take-control-of-your-vcpkg-dependencies-with-versioning-support/). If it is supported, put it also for libcurl.

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #2470

Also, updated dev-time `SHA512 1` to `0` - vcpkg tool now recommends it in the error message, and does not show the expacted SHA with `1` (that was recommended before).

I tested that vcpkg succeeds in my local vcpkg fork - https://github.com/antkmsft/vcpkg/pull/1.

### CODE REVIEW

**Comment by antkmsft** [comment]
@vhvb, when/if Core has 1.1 features and KeyVault needs Core 1.1 and not 1.0, look at the commits 1&2 in this PR to see what needs to be updated.

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.1.0-beta.1 (Unreleased)
+## 1.0.1 (Unreleased)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:3]: Waiting to review and approve this PR until after we have decided what we are doing here.

Comment (antkmsft) [reply sdk/core/azure-core/CHANGELOG.md:3]: We've decided on 1.1, I'll revert the 3rd commit.

File: sdk/core/azure-core/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.1.0-beta.1 (Unreleased)
+## 1.1.0 (Unreleased)
```
Comment (RickWinter) [inline sdk/core/azure-core/CHANGELOG.md:3]: Wait to do this until the point of release

**Review by RickWinter — APPROVED** [review]

File: sdk/core/azure-core/CHANGELOG.md (hunk 2 above)
Comment (antkmsft) [reply sdk/core/azure-core/CHANGELOG.md:3]: We don't have to - when we release, we would also put the release date.

**Pull request merged**